### PR TITLE
Clearer example

### PR DESCRIPTION
**Pull request by tomek-he-him**
Hi Eric!

When I first came here it wasn’t clear to me what this thing does. There was this warm, subconcious feeling that this is totally awesome – but I didn’t quite get the gist.

I’ve been trying to show the awesomeness in a clear way. Let me know if I can improve it further.

### CODE REVIEW

**Comment by ericelliott** [comment]
Great start. I see some unsupported rtype nesting though. In other words:

``` js
otherParam: [x, y]: number[],
```

should be:

``` js
interface coordinates [x: number, y: number]
```

Now you can use it:

``` js
otherParam: coordinates
```

Or, if you want to allow any array of numbers:

``` js
otherParam: number[]
```

I have a feeling that supporting nesting the way you did it would require more complexity in the parser. Lets get it working before we add features to it. ;)

**Comment by ericelliott** [comment]
BTW, if you want to see an example that follows the current spec in working code:

https://github.com/ericelliott/checkin/blob/13cd6ce0ed028c5431409f8981005df2b06c3638/source/shared/store/index.js#L41

**Comment by tomek-he-him** [comment]
Wow! Does [type checking](https://github.com/ericelliott/checkin/blob/13cd6ce0ed028c5431409f8981005df2b06c3638/source/shared/store/index.js#L43) already work there?

> Lets get it working before we add features to it. ;)

Yeah, definitely! I’ll come back here on Monday, I’m quite short on time at the moment. But I’ll update that.

**Comment by tomek-he-him** [comment]
Hi again!

I’ve rebased this onto #12 to keep the example up to date with new conventions (https://github.com/ericelliott/rtype/issues/13).

I wanted to keep the signature as compact as possible for stronger contrast against jsDoc. So I ended up with

``` js
otherParam: [x: Number, y: Number, z: Number]
```

vs

```
 * @param {array} otherParam
 * @param {number} otherParam[0]
 * @param {number} otherParam[1]
 * @param {number} otherParam[2]
```

File: README.md [hunk 1]
```diff
@@ -59,12 +86,18 @@ npm install --save rfx
...
 import rfx from 'rfx';
 
-const predicate = (foo) => {
-  return typeof 'foo' === string;
-};
+export const myFunction = (param, otherParam, options) => {
+  /* (Just business logic.) */
+}
+
+const type = (param, otherParam, options) => (
```
Comment (ericelliott) [inline README.md:93]: Maybe rename this to `MyType`?

**Pull request merged**